pinterest/teletraan
# Prompt User if there is no Access Role on cluster creation

Definition of Done:

Check if there is an access role and prompt if there is no access role on creation
Verify no cluster have blank access_role in prod db
Move default access role to Basic Settings instead of Advanced Settings
Add tooltip telling the what access role to use

Background:
new clusters seem to already have `access_role` blank as a default. It does not prevent users from creating a cluster

Solution:
Created a field in Basic Settings that populates access_role in Advanced Settings
Added the ability for access_role in advanced to populate Access Role in Basic Settings
Added validation for access role with guidance via tooltip

Testing Steps:
1. Navigate to the Capacity page and click Create Cluster
2. On the Create Cluster page, click the tooltip next to Access Role 
<img width="1020" alt="Screen_Shot_2022-10-17_at_11_24_32_AM" src="https://user-images.githubusercontent.com/69278532/196272830-84966ad2-06d6-4e1e-869e-225b7e292c00.png">
4. Confirm that a dropdown message is displayed under the field and that the hyperlink works
5. Populate Access Role with a value
<img width="1726" alt="Screen_Shot_2022-10-17_at_11_23_15_AM" src="https://user-images.githubusercontent.com/69278532/196272903-6ad87282-74a2-4029-a95a-64464b274b57.png">
7. Switch to Advanced Settings and confirm that Access Role disappears and access_role is populated with the value from step  4
<img width="866" alt="Screen_Shot_2022-10-17_at_11_23_36_AM" src="https://user-images.githubusercontent.com/69278532/196272955-c89bf464-0d35-4315-99cd-c160e8f97272.png">
9. Populate access_role with a new value
<img width="867" alt="Screen_Shot_2022-10-17_at_11_23_54_AM" src="https://user-images.githubusercontent.com/69278532/196273039-4945a917-b11e-476c-bcd3-f649840d223a.png">
11. Return to Basic Settings and confirm that Access Role is visible again and is populated with the value from step 6
<img width="865" alt="Screen_Shot_2022-10-17_at_11_24_17_AM" src="https://user-images.githubusercontent.com/69278532/196273072-69df433d-b822-4aa4-933f-b46cdb87a1f0.png">
